internal/run/studio/launchStudio.go

```diff
@@ -0,0 +1,172 @@
+package studio
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/pkg/browser"
+	"github.com/speakeasy-api/speakeasy-core/auth"
+	"github.com/speakeasy-api/speakeasy-core/errors"
+	"github.com/speakeasy-api/speakeasy/internal/log"
+	"github.com/speakeasy-api/speakeasy/internal/run"
+	"go.uber.org/zap"
+)
+
+func LaunchStudio(ctx context.Context, workflow *run.Workflow) error {
+	secret := generateSecret()
+
+	if workflow == nil {
+		return errors.New("unable to launch studio without a workflow")
+	}
+
+	handlers := StudioHandlers{Workflow: *workflow}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/health", handler(handlers.health))
+
+	mux.HandleFunc("/source", func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case http.MethodPost:
+			handler(handlers.updateSource)(w, r)
+		case http.MethodGet:
+			handler(handlers.getSource)(w, r)
+		default:
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		}
+	})
+
+	port, err := searchForAvailablePort()
+	if err != nil {
+		return err
+	}
+
+	server := &http.Server{
+		Addr:    fmt.Sprintf(":%d", port),
+		Handler: corsMiddleware(authMiddleware(secret, mux)),
+	}
+
+	serverURL := auth.GetServerURL()
+
+	url := fmt.Sprintf("%s/studio/%d#%s", serverURL, port, secret)
```

Comment:
I actually don't know. Is URL fragment the best move here rather than query param

Reply:
Yes because servers won't see fragments